lib/ansible/modules/cloud/vmware/vmware_get_info.py

```diff
@@ -0,0 +1,162 @@
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_get_info
+short_description: Gathers facts about various VMware object using REST API
+description:
+- This module can be used to gather facts about various VMware object using REST APIs.
+- This module is based on REST API and uses HTTP API connection plugin for persistent connection.
+version_added: '2.9'
+author:
+- Abhijeet Kasurde (@Akasurde)
+notes:
+- Tested on vSphere 6.5, 6.7
+requirements:
+- python >= 2.6
+options:
+  object_type:
+    description:
+    - Type of VMware object.
+    type: str
+    default: datacenter
+    choices: [ datacenter, cluster, datastore, folder, host, network, resource_pool, virtual_machine,
+               content_library, local_library, subscribed_library, content_type, tag, category ]
+  filters:
+    description:
+    - A list of filters to find the given object.
+    - Valid filters for datacenter object type - folders, datacenters, names.
+    - Valid filters for cluster object type - folders, datacenters, names, clusters.
+    - Valid filters for datastore object type - folders, datacenters, names, datastores, types.
+    - Valid filters for folder object type - folders, parent_folders, names, datacenters, type.
+    - Valid filters for host object type - folders, hosts, names, datacenters, clusters, connection_states.
+    - Valid filters for network object type - folders, types, names, datacenters, networks.
+    - Valid filters for resource_pool object type - resource_pools, parent_resource_pools, names, datacenters, hosts, clusters.
+    - Valid filters for virtual_machine object type - folders, resource_pools, power_states, vms, names, datacenters, hosts, clusters.
+    - content_library, local_library, subscribed_library, content_type, tag, category does not take any filters.
+    default: []
+    type: list
+'''
+
+EXAMPLES = r'''
+- name: Get Datacenter object information
+  vmware_get_info:
+    object_type: datacenter
+    filters:
+      - names: Asia-Datacenter1
+  register: datacenter_result
+
+- set_fact:
+    datacenter_obj: "{{ datacenter_result.object_info.value[0]['datacenter'] }}"
+
+- name: Get all clusters from Asia-Datacenter1
+  vmware_get_info:
+    object_type: cluster
+    filters:
+      - datacenters: "{{ datacenter_obj }}"
+  register: clusters_result
+'''
+
+RETURN = r'''
+object_info:
+    description: facts about the given VMware object
+    returned: always
+    type: dict
+    sample: {
+        "value": [
+            {
+                "cluster": "domain-c42",
+                "drs_enabled": false,
+                "ha_enabled": false,
+                "name": "Asia-Cluster1"
+            }
+        ]
+    }
+'''
+
+from ansible.module_utils.connection import Connection
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = dict(
+        object_type=dict(type='str',
+                         choices=['datacenter', 'cluster',
```

Comment:
VMware may add new resources in the future and if we don't hardcode a list, the user will be able to use it without any update of the module. So I would prefer to avoid the `choices` key.